src/reducers/sources.js

```diff
@@ -126,12 +126,12 @@ function update(
 
     case "ADD_TAB":
       return state.merge({
-        tabs: updateTabList({ sources: state }, action.url)
+        tabs: updateTabList(state.sources.tabs, action.url)
       });
 
     case "MOVE_TAB":
       return state.merge({
-        tabs: updateTabList({ sources: state }, action.url, action.tabIndex)
+        tabs: updateTabList(state.sources.tabs, action.url, action.tabIndex)
```

Comment:
Would it be easier if we did `state.sources.tabs || []`?

Reply:
hmm, what's the benefit of that?

Reply:
I (thought I) saw a circle failure due to `updateTabList` returning undefined

Reply:
got it. We might be able to add a sensible default in initializer